FirebaseDatabase/Sources/Api/FIRDatabaseQuery.m

```diff
@@ -202,20 +202,26 @@ - (FIRDatabaseQuery *)queryStartingAfterValue:(id)startAfterValue {
 
 - (FIRDatabaseQuery *)queryStartingAfterValue:(id)startAfterValue
                                      childKey:(NSString *)childKey {
-    if ([self.queryParams.index isEqual:[FKeyIndex keyIndex]] &&
-        childKey != nil) {
-        @throw [[NSException alloc]
-            initWithName:INVALID_QUERY_PARAM_ERROR
-                  reason:@"You must use queryStartingAfterValue: instead of "
-                         @"queryStartingAfterValue:childKey: when using "
-                         @"queryOrderedByKey:"
-                userInfo:nil];
-    }
-    if (childKey == nil) {
-        childKey = [FUtilities maxName];
+    if ([self.queryParams.index isEqual:[FKeyIndex keyIndex]]) {
+        if (childKey != nil) {
+            @throw [[NSException alloc]
+                initWithName:INVALID_QUERY_PARAM_ERROR
+                      reason:
+                          @"You must use queryStartingAfterValue: instead of "
+                          @"queryStartingAfterValue:childKey: when using "
+                          @"queryOrderedByKey:"
+                    userInfo:nil];
+        }
+        if ([startAfterValue isKindOfClass:[NSString class]]) {
+            startAfterValue = [FNextPushId successor:startAfterValue];
+        }
```

Comment:
What happens when this is not true? Looks like we just call `queryStartingAtInternal` with the original value, which doesn't seem to do any validation.

Reply:
This case is covered by `validateQueryEndpointsForParams`:

https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseDatabase/Sources/Api/FIRDatabaseQuery.m#L106-L107

This is called in `queryingStartingAtInternal`.